### PR TITLE
Remove meteorhacks:meteorx dependency.

### PR DESCRIPTION
**Pull request by benjamn**
This PR brings the relevant bits of [`meteorhacks:meteorx`](https://github.com/meteorhacks/meteorx) into the `mdg:meteor-apm-agent` package, so that [various issues](https://github.com/meteorhacks/meteorx/pull/9) can be fixed without forking or continuing to maintain `meteorhacks:meteorx`.

Related:
https://github.com/meteor/meteor/issues/10337#issuecomment-441717091
https://github.com/meteorhacks/meteorx/issues/8
https://github.com/meteorhacks/meteorx/pull/9
https://github.com/abecks/meteor-fast-render/issues/25

### CODE REVIEW

File: lib/hijack/instrument.js [hunk 1]
```diff
@@ -9,22 +18,22 @@ Kadira._startInstrumenting = function(callback) {
...
   wrapStringifyDDP()
-  MeteorX.onReady(function() {
-    //instrumenting session
-    wrapServer(MeteorX.Server.prototype);
-    wrapSession(MeteorX.Session.prototype);
-    wrapSubscription(MeteorX.Subscription.prototype);
-
-    if(MeteorX.MongoOplogDriver) {
-      wrapOplogObserveDriver(MeteorX.MongoOplogDriver.prototype);
+  Meteor.startup(async function () {
```
Comment (benjamn) [inline lib/hijack/instrument.js:21]: Making this callback function `async` means it will run in a `Fiber`.

File: lib/hijack/meteorx.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
+// Various tricks for accessing "private" Meteor APIs borrowed from the
+// now-unmaintained meteorhacks:meteorx package.
```
Comment (benjamn) [inline lib/hijack/meteorx.js:2]: If you're reviewing this module, you should compare the tricks implemented here to their original implemenations in https://github.com/meteorhacks/meteorx/tree/master/lib.

File: package.js [hunk 3]
```diff
@@ -88,14 +88,23 @@ Package.on_test(function(api) {
...
   if(api.versionsFrom) {
     api.versionsFrom('METEOR@1.2');
-    api.use('meteorhacks:meteorx@1.4.1', ['server']);
     api.use('meteorhacks:zones@1.2.1', {weak: true});
   }
 
   api.use([
-    'minimongo', 'livedata', 'mongo-livedata', 'ejson', 'ddp-common',
-    'underscore', 'http', 'email', 'random'
+    'ecmascript',
```
Comment (benjamn) [inline package.js:95]: Added `ecmascript` so that `lib/hijack/meteorx.js` could be a module with explicit exports.

File: package.js [hunk 4]
```diff
@@ -88,14 +88,23 @@ Package.on_test(function(api) {
...
+    'mongo',
```
Comment (benjamn) [inline package.js:96]: Added `mongo` since it seems to be a dependency of various tricks in the `lib/hijack/meteorx.js` module.

**Review by abernix — APPROVED** [review]
LGTM! Thanks for doing this!

**Review by hwillson — APPROVED** [review]
Looks great @benjamn - thanks!

**Comment by lh0x00** [comment]
I have maintained the [MeteorX](https://github.com/lamhieu-vk/meteorx) project, but your way looks great!

**Comment by benjamn** [comment]
These changes have now been published to Atmosphere as `mdg:meteor-apm-agent@3.2.0`.

**Pull request merged**